### PR TITLE
Epub reader

### PR DESCRIPTION
**Pull request by JanAckermann**
## Related Issue

- Fixes <issue_link>

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests
- [ ] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

## Open tasks:

- [ ] ...

### CODE REVIEW

File: packages/web-app-epub-reader/l10n/translations.json
Comment (dschmidt) [inline packages/web-app-epub-reader/l10n/translations.json:1]: Rather init with an empty object?

File: packages/web-app-epub-reader/src/App.vue [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    },
+    isReadOnly: { type: Boolean, required: false },
+    resource: { type: Object as PropType<Resource>, required: true }
+  },
+  setup(props) {
+    const bookContainer = ref<VNodeRef>()
+
+    const blob: any = new Blob([props.currentContent], { type: 'application/epub+zip' })
+
+    console.log(blob)
```
Comment (dschmidt) [inline packages/web-app-epub-reader/src/App.vue:28]:
```suggestion
```

Comment (dschmidt) [reply packages/web-app-epub-reader/src/App.vue:28]:
```suggestion
```

File: packages/web-app-epub-reader/src/App.vue [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      type: String,
+      required: true
+    },
+    isReadOnly: { type: Boolean, required: false },
+    resource: { type: Object as PropType<Resource>, required: true }
+  },
+  setup(props) {
+    const bookContainer = ref<VNodeRef>()
+
+    const blob: any = new Blob([props.currentContent], { type: 'application/epub+zip' })
```
Comment (dschmidt) [inline packages/web-app-epub-reader/src/App.vue:26]: Why any?

File: packages/web-app-epub-reader/src/App.vue [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    return {
+      bookContainer
+    }
+  }
+})
+</script>
+
+<style lang="scss">
+.oc-text-editor-readonly {
+  //Toastui Editor doesn't have margins in view mode, adjusted for uniformity
```
Comment (dschmidt) [inline packages/web-app-epub-reader/src/App.vue:45]: This isn't toastui editor

File: packages/web-app-epub-reader/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
+import { useGettext } from 'vue3-gettext'
+import translations from '../l10n/translations.json'
+import EPUBReader from './App.vue'
```
Comment (dschmidt) [inline packages/web-app-epub-reader/src/index.ts:3]:
```suggestion
import EpubReader from './App.vue'
```

File: packages/web-app-epub-reader/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    const routes = [
+      {
+        path: '/:driveAliasAndItem(.*)?',
+        component: AppWrapperRoute(EPUBReader, {
+          applicationId: appId
+        }),
+        name: 'epub-reader',
+        meta: {
+          authContext: 'hybrid',
+          title: $gettext('EPUB Reader'),
```
Comment (dschmidt) [inline packages/web-app-epub-reader/src/index.ts:56]:
```suggestion
          title: $gettext('Epub Reader'),
```

File: packages/web-app-epub-reader/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+          patchCleanPath: true
+        }
+      }
+    ]
+
+    return {
+      appInfo: {
+        name: $gettext('Text Editor'),
```
Comment (dschmidt) [inline packages/web-app-epub-reader/src/index.ts:64]: Nope

File: packages/web-app-epub-reader/src/index.ts [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+        id: appId,
+        icon: 'file-text',
+        color: '#0D856F',
+        isFileEditor: true,
+        defaultExtension: 'epub',
+        extensions: fileExtensions().map((extensionItem) => {
+          return {
+            extension: extensionItem.extension,
+            ...(Object.prototype.hasOwnProperty.call(extensionItem, 'newFileMenu') && {
```
Comment (dschmidt) [inline packages/web-app-epub-reader/src/index.ts:73]: Object.hasOwn

File: packages/web-app-epub-reader/tests/unit/app.spec.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
+import { Resource } from '@ownclouders/web-client/src'
+import { AppConfigObject } from '@ownclouders/web-pkg'
+import { mount } from 'web-test-helpers'
+import App from '../../src/App.vue'
+
+vi.mock('@ownclouders/web-pkg')
+
+describe('Text editor app', () => {
```
Comment (dschmidt) [inline packages/web-app-epub-reader/tests/unit/app.spec.ts:8]: Nope

File: packages/web-app-epub-reader/src/App.vue [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+        })
+      },
+      {
+        immediate: true
+      }
+    )
+
+    const navigateLeft = () => {
+      rendition.prev()
+      setNavigateButtonState()
```
Comment (JanAckermann) [inline packages/web-app-epub-reader/src/App.vue]: Remove !

File: packages/web-app-epub-reader/src/App.vue [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+          // Might be sub nav item and therefore undefined
+          if (navItem) {
+            currentChapter.value = navItem
+          }
+
+          navigateLeftDisabled.value = rendition.currentLocation().atStart === true
+          navigateRightDisabled.value = rendition.currentLocation().atEnd === true
+        })
+
+        rendition.on('keydown', (event: Event) => {
```
Comment (JanAckermann) [inline packages/web-app-epub-reader/src/App.vue:107]: @dschmidt any idea to not do this? we already have keydown listeners, and actually bubbeling the event up would be sufficient

**Comment by JanAckermann** [comment]
Known issues:
- esc doesn't close the reader

TBD in follow-ups:
- Add to ocis config.json
- Init transifex 
- Add text size slider
- remember location on reopen

**Review by kulmann — APPROVED** [review]

**Pull request merged**